### PR TITLE
Dedicated docs page for distributed checkpoints

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Creates a dedicated docs page for distributed checkpoints. The feature is currently only briefly explained in our FSDP guide, but with the introduction of #19213 it seems better to have a dedicated page about this topic in detail.

Once reviewed, I will add a very similar page for the Trainer docs.

----
📚 Documentation preview 📚: https://pytorch-lightning--19287.org.readthedocs.build/en/19287/

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

File: docs/source-fabric/guide/checkpoint/distributed_checkpoint.rst [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+        my-checkpoint.ckpt/
+        ├── __0_0.distcp
+        ├── __1_0.distcp
+        ├── __2_0.distcp
+        ├── __3_0.distcp
+        └── meta.pt
+
+    The ``.distcp`` files contain the tensor shards from each process/GPU. You can see that the size of these files
+    is roughly 1/4 of the total size of the checkpoint since the script distributes the model across 4 GPUs.
```
Comment (carmocca) [inline docs/source-fabric/guide/checkpoint/distributed_checkpoint.rst:119]: I would merge these lines with https://github.com/Lightning-AI/pytorch-lightning/pull/19287/files#diff-fc572d5b55ae0e76654f166b314de94f45a5035a94d64be6489dab2eb3343e14R52-R59

File: docs/source-fabric/guide/checkpoint/distributed_checkpoint.rst [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+    # 3. Load using Fabric's method
+    fabric.load("path/to/checkpoint/file", state)
+
+    # DON'T do this (inefficient):
+    # model.load_state_dict(torch.load("path/to/checkpoint/file"))
+
+Note that you can load the distributed checkpoint even if the world size has changed, i.e., you are running on a different number of GPUs than when you saved the checkpoint.
+
+.. collapse:: Full example
```
Comment (carmocca) [inline docs/source-fabric/guide/checkpoint/distributed_checkpoint.rst:153]: This full example is not much longer. You might want to merge it

File: docs/source-fabric/guide/checkpoint/distributed_checkpoint.rst [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+    .. code-block:: python
+
+        import torch
+        import torch.nn.functional as F
```
Comment (carmocca) [inline docs/source-fabric/guide/checkpoint/distributed_checkpoint.rst:158]: Unused

```suggestion
```

**Review by carmocca — APPROVED** [review]

File: docs/source-fabric/guide/checkpoint/index.rst [hunk 4]
```diff
@@ -0,0 +1,32 @@
+:orphan:
```
Comment (Borda) [inline docs/source-fabric/guide/checkpoint/index.rst:1]: if we list it in Glossary, would be easier...

Comment (awaelchli) [reply docs/source-fabric/guide/checkpoint/index.rst:1]: It is linked in the Glossary:

```
.. displayitem::
    :header: Checkpoint
    :button_link: ../guide/checkpoint/index.html
    :col_css: col-md-4
```

**Review by Borda — APPROVED** [review]

File: docs/source-fabric/guide/checkpoint/index.rst (hunk 4 above)
Comment (Borda) [reply docs/source-fabric/guide/checkpoint/index.rst:1]: yes, but this is HTML not the source document that is why you need to add this tag `:orphan:` so I may suggest having both... so when it is in any tree it is a bit harder to lose it :)

Comment (awaelchli) [reply docs/source-fabric/guide/checkpoint/index.rst]: I added it now

**Pull request merged**